### ISSUE TITLE
I got some warning is : Warning: handler '{function_name}' returned a promise and also use a callback! This is problematic and might cause issues in you lambda.

### ISSUE THREAD

**Issue by thiennv95** [spring-media/aws-lambda-router#17]

**Comment by chgohlke** [comment]
Hi @thiennv95, which version of *aws-lambda-router* do you use?

**Comment by thiennv95** [comment]
@chgohlke  my version use is "aws-lambda-router": "^0.5.0",
I just use npm install aws-lambda-router.

**Comment by thiennv95** [comment]
I fixed it with this code : 
In index.js your module. How do you think about this ?
```

"use strict";

function handler(routeConfig) {
    const eventProcessorMapping = extractEventProcessorMapping(routeConfig);

    return async (event, context, callback) => {
        if (routeConfig.debug) {
            console.log("Lambda invoked with request:", event);
            console.log("Lambda invoked with context:", context);
        }

        for (const eventProcessorName of eventProcessorMapping.keys()) {

            try {
                // the contract of 'processors' is as follows:
                // - their method 'process' is called with (config, event)
                // - the method...
                //   - returns null: the processor does not feel responsible for the event
                //   - throws Error: the 'error.toString()' is taken as the error message of processing the event
                //   - returns object: this is taken as the result of processing the event
                //   - returns promise: when the promise is resolved, this is taken as the result of processing the event
                const result = await eventProcessorMapping.get(eventProcessorName)(routeConfig[eventProcessorName], event, context);
                if (result) {
                    // be resilient against a processor returning a value instead of a promise:
                    return result
                } else {
                    if (routeConfig.debug) {
                        console.log("Event processor couldn't handle request.")
                    }
                }
            } catch (error) {
                if (error.stack) {
                    console.log(error.stack);
                }
                return error.toString()
                // callback(error.toString());
                // return;
            }
        }
        return 'No event processor found to handle this kind of event!'
        // callback('No event processor found to handle this kind of event!');
    }
}

function extractEventProcessorMapping(routeConfig) {
    const processorMap = new Map();
    for (let key of Object.keys(routeConfig)) {
        try {
            processorMap.set(key, require(`./lib/${key}`));
        } catch (error) {
            throw new Error(`The event processor '${key}', that is mentioned in the routerConfig, cannot be instantiated (${error.toString()})`);
        }
    }
    return processorMap;
}

module.exports = { handler: handler };
```

**Comment by tdt17** [comment]
Hi @thiennv95,
with our new version 0.8.1 this should be fixed (we use only promises since 0.7.0).
Thanks for your help 👍

**Comment by chgohlke** [comment]
I close this issue, pls use Verion 08.1 or higher. @thiennv95 If you think this dont solve your issue, pls reopen and comment.
